xmtp_id/src/associations/mod.rs

```diff
@@ -1,8 +1,608 @@
+mod association_log;
 mod hashes;
 mod member;
+mod signature;
 mod state;
 #[cfg(test)]
 mod test_utils;
 
+pub use self::association_log::*;
 pub use self::member::{Member, MemberIdentifier, MemberKind};
+pub use self::signature::{Signature, SignatureError, SignatureKind};
 pub use self::state::AssociationState;
+
+// Apply a single IdentityUpdate to an existing AssociationState
+pub fn apply_update(
+    initial_state: AssociationState,
+    update: IdentityUpdate,
+) -> Result<AssociationState, AssociationError> {
+    update.update_state(Some(initial_state))
+}
+
+// Get the current state from an array of `IdentityUpdate`s. Entire operation fails if any operation fails
+pub fn get_state(updates: Vec<IdentityUpdate>) -> Result<AssociationState, AssociationError> {
+    let new_state = updates.iter().try_fold(
+        None,
+        |state, update| -> Result<Option<AssociationState>, AssociationError> {
+            let updated_state = update.update_state(state)?;
+            Ok(Some(updated_state))
+        },
+    )?;
+
+    new_state.ok_or(AssociationError::NotCreated)
+}
+
+#[cfg(test)]
```

Comment:
It'd be nice to add a summary for the test suite - type/order of things to be tested.

Reply:
That's a good idea. Once the XIP gets published, maybe I can just add a link to the relevant section. That definitely informed a lot of the test cases.